app/javascript/Domain/UseCases/Nations/GiveTurn.js

```diff
@@ -0,0 +1,41 @@
+export default class GiveTurn {
+  #game;
+
+  constructor(game) {
+    this.#game = game;
+  }
+
+  get #atLeastOneNationHasGovernor() {
+    for (const nation of this.#game.allNations()) {
+      if (nation.governor) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  giveTurnToNextGovernedNation(undoHistory) {
+    if (this.#atLeastOneNationHasGovernor === false) {
+      return;
+    }
+
+    let nextGovernedNation = this.#game.currentNation;
+    do {
+      nextGovernedNation = this.#game.nationTurnAfter(nextGovernedNation);
+    } while (nextGovernedNation.governor === null);
+
+    this.forceGiveTurnTo(nextGovernedNation, undoHistory);
+  }
+  giveTurnToNextNation(undoHistory) {
+    this.forceGiveTurnTo(this.#game.nationTurnAfter(this.#game.currentNation), undoHistory);
+  }
+  forceGiveTurnTo(nation, undoHistory) {
```

Comment:
Honestly most of this particular file is reworked in the next PR due to changes in the underlying `Nation` entity, won't worry too much about it